src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestHeaders.cs

```diff
@@ -21,11 +21,11 @@ public Enumerator GetEnumerator()
 
         public partial struct Enumerator : IEnumerator<KeyValuePair<string, StringValues>>
         {
-            private FrameRequestHeaders _collection;
-            private long _bits;
+            private readonly FrameRequestHeaders _collection;
+            private readonly long _bits;
             private int _state;
             private KeyValuePair<string, StringValues> _current;
-            private bool _hasUnknown;
+            private readonly bool _hasUnknown;
             private Dictionary<string, StringValues>.Enumerator _unknownEnumerator;
```

Comment:
Have to be careful with `readonly` of in a struct as the compiler causes a defensive copy to be taken to prevent mutation so you don't modify the original which will be why making the `_unknownEnumerator` readonly will cause the tests to fail (as the enumerator needs to mutate) is ok in classes. Ref: http://codeblog.jonskeet.uk/2014/07/16/micro-optimization-the-surprising-inefficiency-of-readonly-fields/